src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxValueProvider_ForAttributeWithSimpleName.cs

```diff
@@ -77,9 +82,8 @@ public partial struct SyntaxValueProvider
             .WithTrackingName("allUpIncludingCompilationGlobalAliases_ForAttribute");
 
         // Create a syntax provider for every compilation unit.
-        var compilationUnitProvider = this.CreateSyntaxProvider(
-            static (n, _) => n is ICompilationUnitSyntax,
-            static (context, _) => context.Node).WithTrackingName("compilationUnit_ForAttribute");
+        var compilationUnitProvider = syntaxTreesProvider.Select(
+            (t, c) => t.GetRoot(c)).WithTrackingName("compilationUnit_ForAttribute");
```

Comment:
static?
```suggestion
            static (t, c) => t.GetRoot(c)).WithTrackingName("compilationUnit_ForAttribute");
```

Reply:
actually, noticing this, this is bad.  we don't want to hold onto the roots of all the trees (as that will prevent them from being GC'ed).  moving this to only hold the syntax tree, and to delay getting the root until necessary.